iceoryx_hoofs/utility/include/iox/detail/convert.inl

```diff
@@ -58,384 +60,404 @@ convert::toString(const Source& t) noexcept
 }
 
 template <typename Destination>
-inline bool convert::fromString(const char* v, Destination& dest) noexcept
+inline iox::optional<Destination> convert::from_string(const char* v) noexcept
 {
-    dest = Destination(v);
-    return true;
+    if constexpr (is_iox_string<Destination>::value)
+    {
+        using IoxString = Destination;
+        if (strlen(v) > IoxString::capacity())
+        {
+            return iox::nullopt;
+        }
+        return iox::optional<IoxString>(IoxString(TruncateToCapacity, v));
+    }
+    else
+    {
+        static_assert(always_false_v<Destination>,
+                      "For a conversion to 'std::string' please include 'iox/std_string_support.hpp'!\nConversion not "
+                      "supported!");
+    }
 }
 
 template <>
-inline bool convert::fromString<char>(const char* v, char& dest) noexcept
+inline iox::optional<char> convert::from_string<char>(const char* v) noexcept
 {
     if (strlen(v) != 1U)
     {
         IOX_LOG(DEBUG, v << " is not a char");
-        return false;
+        return iox::nullopt;
     }
 
     /// @NOLINTJUSTIFICATION encapsulated in abstraction
     /// @NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
-    dest = v[0];
-    return true;
+    return iox::optional<char>(v[0]);
 }
 
-template <uint64_t Capacity>
-inline bool convert::fromString(const char* v, string<Capacity>& dest) noexcept
+template <>
+inline iox::optional<bool> convert::from_string<bool>(const char* v) noexcept
 {
-    if (strlen(v) > Capacity)
+    // we should clean errno first
+    errno = 0;
+    char* end_ptr = nullptr;
+
+    if (start_with_neg_sign(v))
     {
-        return false;
+        return iox::nullopt;
     }
 
-    dest = string<Capacity>(TruncateToCapacity, v);
-    return true;
+    auto call = IOX_POSIX_CALL(strtoul)(v, &end_ptr, STRTOUL_BASE)
+                    .failureReturnValue(ULONG_MAX)
+                    .suppressErrorMessagesForErrnos(EINVAL, ERANGE)
+                    .evaluate();
+
+    // we assume that in the IOX_POSIX_CALL procedure, no other POSIX call will change errno,
+    // except for the target function 'f'.
+    return evaluate_return_value<bool>(call, errno, end_ptr, v);
```

Comment:
Oh, I realized just now that you used the `errno` directly. The `IOX_POSIX_CALL` returns an `expected<PosixCallResult<T>, PosixCallResult<T>>` which already has this information. You therefore can access the `errno` via `call.value().errnum` if the call was successful or via `call.error().errnum` if the call failed. Similar with the actual return value, `call.value().value` if successful or `call.error().value` if the call failed.

Could you also change this? You basically just need to remove the `errno` parameter from `evaluate_return_value`and then use `call->errnum` to pass it to `check_edge_case`.